packages/kbn-securitysolution-autocomplete/src/field_value_match/index.tsx

```diff
@@ -214,7 +214,10 @@ export const AutocompleteFieldMatchComponent: React.FC<AutocompleteFieldMatchPro
     if (onError != null) onError(false);
 
     handleSpacesWarning(selectedValue);
-  }, [selectedField, selectedValue, handleSpacesWarning, onError]);
+    // Looks like selectedField return new object every time when we for example add "and" entry
+    // that's why we need to check for name and type here
+    // Probably we should use some kind of memoization on parent components for entries
+  }, [selectedField?.name, selectedField?.type, selectedValue, handleSpacesWarning, onError]);
```

Comment:
These comments are really insightful thanks for finding the root cause of this. Would it be possible for you to add these comments to an issue? This way we can try to add it to the next release to-do list?

Reply:
Sure, will create an issue. This comment fixes one of the reasons of the validation problem, but main cause it's how we stored our errors before

Reply:
Here the issue: [](https://github.com/elastic/kibana/issues/151735)